### DISCUSSION TITLE
[Feature Request]: Power Saving Feature - Reduce LoRa TX power based on dusk/dawn-times or USB==1/0

### DISCUSSION THREAD

**Discussion by h0lad** [meshtastic/firmware#4616] [Ideas]
### Platform

NRF52, ESP32, RP2040, Linux Native, other

### Description

A few solar charge controllers (TI BQ25185/etc.) support a Power Path'ing which means that the device can be powered directly from the solar panel as soon as the battery is fully charged.  It also mixes (70% solar, 30% battery)  the power  during dusk/dawn or cloudy phases.

Example, I marked the spots where the device switches to direct pass through:

![image](https://github.com/user-attachments/assets/56b1437d-7ceb-4f52-a607-6e7c94217ae6)

Feature Ideas: It would be great to have an automata which reduces the TX power to a defined level if  a specific condition is met.

1.  USB == 1 / 0 might be an indicator. A slowly following switching logic would work with the TI power path functionality. 

3. As we  introduced a  system time lately we could even make this with calculated dusk/dawn times based on GPS coordinates 

4. Or a threshold-driven approach: reduce TX power by -10dB if we are reaching 50% battery

**Comment by b8b8** [comment]
What would be the situation where you would want lower transmit power but not say just turning the transmitter off completely when a certain battery threshold is met? Does the intermittent transmission or relay of messages significantly effect the battery life compared to the CPU and other peripherals running? I can see turning the transmission to off at say 10%, but 50% battery could still be 2 weeks of power on an NRF device. I guess my question is what is the use of a lower power transmit unless you are using one of the multi-watt station G2 edition devices.

**Comment by h0lad** [comment]
> What would be the situation where you would want lower transmit power but not say just turning the transmitter off completely when a certain battery threshold is met? 

Full TX power during sunrise until sunset as the device most likely can be directly powered from the solar panel - even while recharging the battery. Reduced during the nighttime.

The LoRa transmitter of your NRF will also eat quite a bit of energy. The uC is not a relevant factor. Peak current according to Semtec (+20dB + PA-Boost) you can plan for ~120mA peak. With only +14dBm it's only ~30mA.

![image](https://github.com/user-attachments/assets/13c40e7b-653d-484a-aaad-2232f85fbdf7)

**Comment by b8b8** [comment]
ok, then i ask is it good to have a node that may only be reachable at some times of the day knowing that further away nodes wouldnt be able to communicate with it at night? There would also be less traffic at night, meaning less packets transmitted and draining the battery. 

This is an interesting idea, i just trying to see the point of lowering power at all if you are not near a completely depleted battery and the next day it should charge up a bit.

**Comment by h0lad** [comment]
During the summer it's most likely no problem to keep it running on full TX power the entire time as you going to reach 100% SoC every mid day.  But during the winter and autumn it can give you a lot of margin to keep your node running when there's bad weather around.

Also you can build stations with smaller solar cells and batteries if you can manage to reduce the power consumption dynamically.

**Comment by b8b8** [comment]
would be interesting to run an experiment to see 2 identical nodes, one set for 10dbm transmit and one set for 22dbm and measure how much of a difference there is over a week.

**Comment by h0lad** [comment]
I will start that experiment! Have to build two stations anyway and then will have to "burn them in" for 1-2 weeks. Going to set one to maximum power and one to +12dBm.

Just waiting for the 18650 battery holders to arrive to solder them. They are already somewhere in a central post station.

**Comment by h0lad** [comment]
![image](https://github.com/user-attachments/assets/61fdc3c0-e40e-4cfd-a385-77e9cec4ef1b)

First measurements:
 - HT-CT62, Router mode, Minimum Wake interval 1s, CPU 80MHz
 - 290mA, 6V panel
 - 4000mAh 3.7V LiPo, 4.1v charge cut-off voltage
 - 27dBm TX Power

Jumps >4.2V = pass-through mode, meaning battery is full

Now switched to 10dBm TX power.

**Comment by h0lad** [comment]
Measurements are still in progress. First  reduction to +10dBm TX didn't show a major improvement.. looks like the ESP32 is still the biggest power drainer.

Maybe somebody with a NRF (RAK, etc.) can test this. First do a run with full legal LoRa TX power and then a second one with only +10dBm.

**Reply by b8b8** [reply]
I have 2 NRF T1000-E's. I can charge them both fully, set one to 10dbm and one to 22dbm and test over 24 hours to see the difference. I have a very chatty local mesh so this should be a good test. Devices are both reprogrammed now and power settings for one lowered to 10. I am just topping up the batteries and will run the test for 24 hours.

**Comment by b8b8** [comment]
Some data from the last day.
![image](https://github.com/user-attachments/assets/ac0bf419-698a-4e44-8efe-d73b5158123a)

**Reply by h0lad** [reply]
Thank you! Looks like my feature request is not worth the hassle.. we are saving 1-3% by reducing the TX power of the station.

Maybe the effect will be bigger with a heavily loaded router  station but not for the standard ones.

**Reply by b8b8** [reply]
Happy to do the test. This idea was def worth the investigation. Ill just add my telemetry settings, etc.

**Reply by h0lad** [reply]
Thanks! I'm closing this discussion.

**Reply by b8b8** [reply]
![image](https://github.com/user-attachments/assets/dbdf2a0f-7998-4f50-b9dc-dfdc5afd2c00)

**Comment by h0lad** [comment]
TX power reduction to 10dBm doesn't reduce the power consumption enough to extend the lifetime of a device in an useable way. Maybe it's that idle RX power consumption of the LoRa chipset that we cannot get rid of. Closing this feature request.

**Comment by FFAMax** [comment]
Thanks @dm5tt for this topic. A year later I found the same feature useful.

Another request was opened https://github.com/meshtastic/firmware/issues/7884 
If properly implement all pieces it will make sense.
My topic was raised after working in real environment far away from electricity with low Lora coverage.
The biggest issue - battery may be drained completely without a chance to use it in deep sleep for emergency or urgent messages (retransmitter on the mountain, etc.)
Need improve power efficiency further.

**Reply by h0lad** [reply]
With Lora the lower limit of the power efficiency always will be determined by the idle RX current of the module: it will never be below 5-6mA on the newer ones or 10-11mA on the older ones. Independent of the MCU.

**Reply by FFAMax** [reply]
> it will never be below 5-6mA on the newer ones or 10-11mA on the older ones. Independent of the MCU.

Yep. That's why, if battery level is low, instead of drain battery in 1h in normal mode, can cut off Lora module, put MCU into sleep mode for 30 minutes then wake up for 5 minute so it will stay reachable for 5 min every 30 minutes, etc.
Instead of just to be dead till next sun activity, etc.
